<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] Cathing broker shut down in a consumer
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Cathing%20broker%20shut%20down%20in%20a%20consumer&In-Reply-To=%3CE74971C5-D73B-40C8-B96B-AE63083C28F6%40rabbitmq.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="024632.html">
   <LINK REL="Next"  HREF="024658.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] Cathing broker shut down in a consumer</H1>
    <B>Tim Watson</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Cathing%20broker%20shut%20down%20in%20a%20consumer&In-Reply-To=%3CE74971C5-D73B-40C8-B96B-AE63083C28F6%40rabbitmq.com%3E"
       TITLE="[rabbitmq-discuss] Cathing broker shut down in a consumer">tim at rabbitmq.com
       </A><BR>
    <I>Fri Dec 21 12:06:21 GMT 2012</I>
    <P><UL>
        <LI>Previous message: <A HREF="024632.html">[rabbitmq-discuss] Cathing broker shut down in a consumer
</A></li>
        <LI>Next message: <A HREF="024658.html">[rabbitmq-discuss] Cathing broker shut down in a consumer
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#24647">[ date ]</a>
              <a href="thread.html#24647">[ thread ]</a>
              <a href="subject.html#24647">[ subject ]</a>
              <a href="author.html#24647">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On 20 Dec 2012, at 18:55, CJ wrote:
&gt;<i> I do not seem to be able to catch the problem of the broker coming down in 
</I>&gt;<i> the consumer.  I use the following code on the nextDelivery
</I>&gt;<i>  
</I>&gt;<i>       catch (ShutdownSignalException ex) {
</I>&gt;<i>          System.out.println(&quot;--- Broker Shut Down ---&quot;);
</I>&gt;<i>       }
</I>&gt;<i>       catch (Exception ex) {
</I>&gt;<i>          System.out.println(&quot;--- Broker Shut Down ---&quot;);
</I>&gt;<i>       }
</I>&gt;<i>  
</I>&gt;<i> But I get the following error:
</I>&gt;<i>  
</I>&gt;<i> Exception in thread &quot;main&quot; com.rabbitmq.client.ShutdownSignalException: connecti
</I>&gt;<i> on error; reason: {#method&lt;connection.close&gt;(reply-code=320, reply-text=CONNECTI
</I>&gt;<i> ON_FORCED - broker forced connection closure with reason 'shutdown', class-id=0,
</I>&gt;<i>  method-id=0), null, &quot;&quot;}
</I>&gt;<i>         at com.rabbitmq.client.QueueingConsumer.handle(QueueingConsumer.java:198
</I>&gt;<i> )
</I>&gt;<i>         at com.rabbitmq.client.QueueingConsumer.nextDelivery(QueueingConsumer.ja
</I>&gt;<i> va:227)
</I>&gt;<i>         at Recv.main(Recv.java:50)
</I>&gt;<i>  
</I>&gt;<i> Does Rabbit support catching broken connections in the consumer ?
</I>&gt;<i> 
</I>
Are you sure you're looking at the exception that you think!? More to the point, are you sure it's being thrown in the thread you think you're catching it in?

</PRE>


























<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="024632.html">[rabbitmq-discuss] Cathing broker shut down in a consumer
</A></li>
	<LI>Next message: <A HREF="024658.html">[rabbitmq-discuss] Cathing broker shut down in a consumer
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#24647">[ date ]</a>
              <a href="thread.html#24647">[ thread ]</a>
              <a href="subject.html#24647">[ subject ]</a>
              <a href="author.html#24647">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
